weatherman.py

```diff
@@ -0,0 +1,189 @@
+        self.max_temperature_date = date
+
+    def set_min_temperature(self, temperature, date):
+        self.min_temperature = temperature
+        self.min_temperature_date = date
+    
+    def set_max_humidity(self, humidity, date):
+        self.max_humidity = humidity
+        self.max_humidity_date = date
+    
+    def get_max_temps(self):
+        return self.max_temperature, self.min_temperature, self.max_humidity
+
+    def print_temperatures_year(self):
+        print(f"highest temperature is: {self.max_temperature} C on {self.max_temperature_date}")
+        print(f"lowest temperature is: {self.min_temperature} C on {self.min_temperature_date}")
+        print(f"highest humidity percentage is: {self.max_humidity} % on {self.max_humidity_date}")
+
+
+temperature_tracker = temperatureTracker()
+
+
+def open_csv_file(file_name):    
+    csv_file = open(file_name) 
+    return csv_file
+
+
+def read_weather_data_from_csv(csv_file):
+    weather_data = list(csv.DictReader(csv_file, skipinitialspace=True))
+    if len(weather_data) == 0:
+        return False
+    return weather_data
+
+
+def match_file_name(year, month_name):                            
+    flag = False    
+    for file_name in os.listdir('.'):
+        if year in file_name and month_name in file_name:
+            return file_name          
+    if not flag:     
+        print("NO RECORD FOUND AGAINT THIS NAME")
+        return False
+
+
+def extract_int_value(value): return int(value) if value != "" else 0
+
+
+def calculate_temprature_sum_for_each_record(weather_data): 
+    sum_of_temperature_values = {"max_temperature": 0, "min_temperature": 0, "mean_humidity": 0}   
+    for line_read in weather_data:                                    
+            sum_of_temperature_values["max_temperature"] += extract_int_value(line_read["Max TemperatureC"])            
+            sum_of_temperature_values["min_temperature"] += extract_int_value(line_read["Min TemperatureC"])
+            sum_of_temperature_values["mean_humidity"] += extract_int_value(line_read["Mean Humidity"] )
+    return  sum_of_temperature_values
+
+
+def calculate_average(sum_of_temperature_values, total_number_of_values):    
+    for key in sum_of_temperature_values:                           
+        sum_of_temperature_values[key] = sum_of_temperature_values[key] / total_number_of_values
+    return sum_of_temperature_values
+
+
+def print_temperatures_month(average_values):
+    print(f"AVG Maximum temperature is: {average_values['max_temperature']} C ")
+    print(f"AVG Minimum temperature is: {average_values['min_temperature']} C ")
+    print(f"AVG Mean humidity percentage is: {average_values['mean_humidity']} %")
+
+    
+def generate_report_for_month_controller(file_name):   
+    csv_file = open_csv_file(file_name)
+    weather_data = read_weather_data_from_csv(csv_file)
+    total_records = len(weather_data)
+    if weather_data == 0: 
+        print(file_name, "File Empty")
```

Comment:
Why does report generation needs to see file is empty or not? Don't you feel that this concern should have been addressed by the reader?